frontend/packages/component-library/src/tags/__tests__/Tags.test.tsx

```diff
@@ -143,4 +143,84 @@ describe('Design System - Tags Component', () => {
     expect(tagLabel).toBeInTheDocument();
     expect(icon).toBeInTheDocument();
   });
+
+  describe('onClose', () => {
+    let user: UserEvent;
+    let onClick1: jest.Mock;
+    let onClick2: jest.Mock;
+    let tagsList: TagProps[];
+    let button1: HTMLElement;
+    let button2: HTMLElement;
+
+    beforeEach(() => {
+      user = userEvent.setup();
+      onClick1 = jest.fn();
+      onClick2 = jest.fn();
+      tagsList = [
+        {
+          tooltipId: 'tag-icon-1',
+          label: 'tag with icon',
+          tooltipContent: 'Tooltip with icon',
+          onClose: onClick1,
+          icon: {
+            iconName: 'close',
+            iconStyle: 'solid',
+            title: 'Close 1',
+            placement: 'right',
+          },
+        },
+        {
+          tooltipId: 'tag-icon-2',
+          label: 'tag with icon',
+          tooltipContent: 'Tooltip with icon',
+          onClose: onClick2,
+          icon: {
+            iconName: 'close',
+            iconStyle: 'solid',
+            title: 'Close 2',
+            placement: 'right',
+          },
+        },
+      ];
+
+      render(<Tags tagsList={tagsList} />);
+
+      [button1, button2] = screen.getAllByRole('button', {
+        name: /close/i,
+      });
+    });
+
+    it('has keyboard navigable tag icons if onClick is present', async () => {
+      await user.tab();
+
+      expect(button1).toHaveFocus();
+
+      await user.tab();
+
+      expect(button2).toHaveFocus();
+    });
+
+    it('calls onClick when an icon is clicked', async () => {
+      await user.click(button1);
+
+      expect(onClick1).toHaveBeenCalledTimes(1);
+      expect(onClick2).not.toHaveBeenCalled();
+    });
+
+    it('calls onClick when Enter or Space is pressed while tag has focus', async () => {
```

Comment:
super nit: might be worth using onClose instead of onClick as it may be misleading